pkg/kubelet/kuberuntime/kuberuntime_manager.go

```diff
@@ -722,52 +730,61 @@ func (m *kubeGenericRuntimeManager) doPodResizeAction(pod *v1.Pod, podStatus *ku
 		if err != nil {
 			klog.ErrorS(err, "GetPodCgroupConfig for memory failed", "pod", pod.Name)
-			result.Fail(err)
+			resizePodResult.Fail(kubecontainer.ErrResizePodSandbox, fmt.Sprintf("get pod cgroup config for memory failed, %s", err.Error()))
 			return
 		}
 		currentPodMemoryUsage, err := pcm.GetPodCgroupMemoryUsage(pod)
 		if err != nil {
 			klog.ErrorS(err, "GetPodCgroupMemoryUsage failed", "pod", pod.Name)
-			result.Fail(err)
+			resizePodResult.Fail(kubecontainer.ErrResizePodSandbox, fmt.Sprintf("get pod cgroup memory usage failed, %s", err.Error()))
 			return
 		}
 		if currentPodMemoryUsage >= uint64(*podResources.Memory) {
 			klog.ErrorS(nil, "Aborting attempt to set pod memory limit less than current memory usage", "pod", pod.Name)
-			result.Fail(fmt.Errorf("Aborting attempt to set pod memory limit less than current memory usage for pod %s", pod.Name))
+			resizePodResult.Fail(kubecontainer.ErrResizePodSandbox, "Aborting attempt to set pod memory limit less than current memory usage")
 			return
 		}
 		if errResize := resizeContainers(v1.ResourceMemory, int64(*currentPodMemoryConfig.Memory), *podResources.Memory, 0, 0); errResize != nil {
-			result.Fail(errResize)
+			resizePodResult.Fail(kubecontainer.ErrResizePodSandbox, fmt.Sprintf("resizeContainers for memory failed, %s", errResize.Error()))
 			return
 		}
 	}
 	if len(podContainerChanges.ContainersToUpdate[v1.ResourceCPU]) > 0 || podContainerChanges.UpdatePodResources {
 		if podResources.CPUQuota == nil || podResources.CPUShares == nil {
 			klog.ErrorS(nil, "podResources.CPUQuota or podResources.CPUShares is nil", "pod", pod.Name)
-			result.Fail(fmt.Errorf("podResources.CPUQuota or podResources.CPUShares is nil for pod %s", pod.Name))
+			resizePodResult.Fail(kubecontainer.ErrResizePodSandbox, "podResources.CPUQuota or podResources.CPUShares is nil")
 			return
 		}
 		currentPodCpuConfig, err := pcm.GetPodCgroupConfig(pod, v1.ResourceCPU)
 		if err != nil {
 			klog.ErrorS(err, "GetPodCgroupConfig for CPU failed", "pod", pod.Name)
-			result.Fail(err)
+			resizePodResult.Fail(kubecontainer.ErrResizePodSandbox, fmt.Sprintf("get pod cgroup config for cpu failed, %s", err.Error()))
 			return
 		}
 		if errResize := resizeContainers(v1.ResourceCPU, *currentPodCpuConfig.CPUQuota, *podResources.CPUQuota,
 			int64(*currentPodCpuConfig.CPUShares), int64(*podResources.CPUShares)); errResize != nil {
-			result.Fail(errResize)
+			resizePodResult.Fail(kubecontainer.ErrResizePodSandbox, fmt.Sprintf("resizeContainers for cpu failed, %s", errResize.Error()))
 			return
 		}
 	}
+
+	for resourceName, cInfos := range podContainerChanges.ContainersToUpdate {
+		for _, cInfo := range cInfos {
+			container := pod.Spec.Containers[cInfo.apiContainerIdx]
+			m.recordContainerEvent(pod, &container, "", v1.EventTypeNormal, events.FinishedVerticalScalingPod, "Pod vertical scaling pod %s resource finished", resourceName)
```

Comment:
This should be recorded after successful  m.updateContainerResources in updatePodContainerResources function.